lib/min_heap.rb

```diff
@@ -8,26 +8,32 @@ def initialize(key, value)
 end
 
 class MinHeap
-
   def initialize
     @store = []
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(log n) where n is the number of elements in the heap
+  # Space Complexity: O(1)
```

Comment:
`heap_up` is recursive and you have to consider that and it's affect on the call stack in terms of space complexity.